app/src/server/api/routers/organizations.router.ts

```diff
@@ -0,0 +1,146 @@
+import { TRPCError } from "@trpc/server";
+import { v4 as uuidv4 } from "uuid";
+import { z } from "zod";
+
+import { createTRPCRouter, protectedProcedure } from "~/server/api/trpc";
+import { prisma } from "~/server/db";
+import { generateApiKey } from "~/server/utils/generateApiKey";
+import {
+  requireCanModifyOrganization,
+  requireIsOrgAdmin,
+  requireNothing,
+} from "~/utils/accessControl";
+
+export const organizationsRouter = createTRPCRouter({
+  list: protectedProcedure.query(async ({ ctx }) => {
+    const userId = ctx.session.user.id;
+    requireNothing(ctx);
+
+    if (!userId) {
+      return null;
+    }
+
+    const organizations = await prisma.organization.findMany({
+      where: {
+        organizationUsers: {
+          some: { userId: ctx.session.user.id },
+        },
+      },
+      orderBy: {
+        createdAt: "asc",
+      },
+    });
+
+    if (!organizations.length) {
+      const newOrgId = uuidv4();
+      const [newOrg] = await prisma.$transaction([
```

Comment:
Is there a reason this really has to exist in the list call? We're going to end up with race conditions where multiple projects are created bc of server-side rendering immediately followed by client-side rendering. If it absolutely has to happen is there any way to do it with an upsert?

Reply:
No it doesn't need to exist in a list call. This is merely a hack to ensure the user automatically gets a project after signing up. A better solution would be to add a new route which specifically checks if the user doesn't already have a project and upserts one in that event. We should then call that route every time a user signs up. I'm not sure how to do that with github auth, but it's probably possible.